presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/ClientInfoFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.verifier.prestoaction;
+
+import com.facebook.presto.verifier.framework.QueryStage;
+import com.facebook.presto.verifier.framework.VerificationContext;
+import com.facebook.presto.verifier.framework.VerifierConfig;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+
+public class ClientInfoFactory
+{
+    private Injector injector;
+
+    @Inject
+    public void setInjector(Injector injector)
+    {
+        this.injector = injector;
```

Comment:
It is usually discouraged to use the `Injector` directly as it may cause various initialization order related problems. From what I remember If the Injector is used directly the `Guice`  framework doesn't know statically what dependency needs to be satisfied via `injector.getInstance` and may initialize instances in a wrong order. For example the framework might decide to initialize `ClientInfoFactory` before the `VerifierConfig` is initialized. 

Currently it is more of a theoretical problem, because `ClientInfoFactory#create` is not called during initialization, but there's a chance someone might decide to use it during initialization causing an execution order dependent failure.

I would consider introducing an `ClientInfoFactory` interface with a single `ClientInfo create(VerificationContext verificationContext, QueryStage queryStage)` method. Then you can return a `Class<? extends ClientInfoFactory>` from the `VerifyCommand#getClientInfoFactory` with the default implementation (e.g.: `DefaultClientInfoFactory.class`). Then have the `VerifierConfig` injected directly into the `DefaultClientInfoFactory` constructor.

Reply:
Thanks for the review, yeh it makes sense to use @inject feature instead of using injector.getInstance() in create method to ensure initializing order. 
i also make clientinfoFactory as a singleton